drivers/sensor/apds9253/apds9253.c

```diff
@@ -0,0 +1,260 @@
+		return -EIO;
+	}
+
+	LOG_DBG("status: 0x%x", tmp);
+
+	if (tmp & APDS9253_MAIN_STATUS_LS_STATUS) {
+		if (i2c_burst_read_dt(&config->i2c, APDS9253_LS_DATA_BASE, buf, sizeof(buf))) {
+			return -EIO;
+		}
+
+		for (int i = 0; i < VALUES_PER_SAMPLE; ++i) {
+			data->sample_crgb[i] = get_value_from_buf(i, buf);
+		}
+
+		LOG_DBG("IR 0x%x GREEN 0x%x BLUE 0x%x RED 0x%x\n", data->sample_crgb[0],
+			data->sample_crgb[1], data->sample_crgb[2], data->sample_crgb[3]);
+	}
+
+	return 0;
+}
+
+static int apds9253_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct apds9253_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_IR:
+		val->val1 = sys_le32_to_cpu(data->sample_crgb[0]);
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_GREEN:
+		val->val1 = sys_le32_to_cpu(data->sample_crgb[1]);
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_BLUE:
+		val->val1 = sys_le32_to_cpu(data->sample_crgb[2]);
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_RED:
+		val->val1 = sys_le32_to_cpu(data->sample_crgb[3]);
+		val->val2 = 0;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int apds9253_sensor_setup(const struct device *dev)
+{
+	const struct apds9253_config *config = dev->config;
+	uint8_t chip_id;
+
+	if (i2c_reg_read_byte_dt(&config->i2c, APDS9253_PART_ID, &chip_id)) {
+		LOG_ERR("Failed reading chip id");
+		return -EIO;
+	}
+
+	if (!((chip_id & APDS9253_PART_ID_ID_MASK) == APDS9253_DEVICE_PART_ID)) {
+		LOG_ERR("Invalid chip id 0x%x", chip_id);
+		return -EIO;
+	}
+
+	/* LS Gain */
+	if (i2c_reg_update_byte_dt(&config->i2c, APDS9253_LS_GAIN_REG, APDS9253_LS_GAIN_MASK,
+				   (config->ls_gain & APDS9253_LS_GAIN_MASK))) {
+		LOG_ERR("Light sensor gain is not set");
+		return -EIO;
+	}
+
+	/* LS Resolution */
+	if (i2c_reg_update_byte_dt(&config->i2c, APDS9253_LS_MEAS_RATE_REG,
+				   APDS9253_LS_MEAS_RATE_RES_MASK,
+				   (config->ls_resolution & APDS9253_LS_MEAS_RATE_RES_MASK))) {
+		LOG_ERR("Light sensor resolution is not set");
+		return -EIO;
+	}
+
+	/* LS Rate */
+	if (i2c_reg_update_byte_dt(&config->i2c, APDS9253_LS_MEAS_RATE_REG,
+				   APDS9253_LS_MEAS_RATE_MES_MASK,
+				   (config->ls_rate & APDS9253_LS_MEAS_RATE_MES_MASK))) {
+		LOG_ERR("Light sensor rate is not set");
+		return -EIO;
+	}
+
+	/* Enable RGB mode */
+	if (i2c_reg_update_byte_dt(&config->i2c, APDS9253_MAIN_CTRL_REG,
+				   APDS9253_MAIN_CTRL_RGB_MODE, APDS9253_MAIN_CTRL_RGB_MODE)) {
+		LOG_ERR("Enable RGB mode");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int apds9253_init_interrupt(const struct device *dev)
+{
+	const struct apds9253_config *config = dev->config;
+	struct apds9253_data *drv_data = dev->data;
+
+	if (!gpio_is_ready_dt(&config->int_gpio)) {
+		LOG_ERR("%s: device %s is not ready", dev->name, config->int_gpio.port->name);
+		return -ENODEV;
+	}
+
+	gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT | config->int_gpio.dt_flags);
```

Comment:
`gpio_pin_configure_dt` already uses `dt_flags`
```suggestion
	gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
```
Also it might be a good idea to check the return value of ``gpio_pin_configure_dt`.